sdk/js/src/api/http.js

```diff
@@ -15,32 +15,55 @@
 import axios from 'axios'
 import { cloneDeep, get, isObject } from 'lodash'
 
-import { URI_PREFIX_STACK_COMPONENT_MAP, STACK_COMPONENTS_MAP } from '../util/constants'
+import Token from '../util/token'
 import EventHandler from '../util/events'
+import {
+  URI_PREFIX_STACK_COMPONENT_MAP,
+  STACK_COMPONENTS_MAP,
+  AUTHENTICATION_MODES,
+} from '../util/constants'
 
 import stream from './stream/stream-node'
 
 /**
  * Http Class is a connector for the API that uses the HTTP bridge to connect.
  */
 class Http {
-  constructor(token, stackConfig, axiosConfig = {}) {
+  constructor(authentication, stackConfig, axiosConfig = {}) {
+    let authToken
+    let csrfToken
+    if (typeof authentication === 'object') {
```

Comment:
It is very hard to process this. Please consider simplifying and removing 3 nested ifs:

```diff
@ sdk/js/src/api/http.js:33
 class Http {
   constructor(authentication, stackConfig, axiosConfig = {}) {
+    if (typeof authentication !== 'object') {
+      throw new Error('kek')
+    }
+    this._stackConfig = stackConfig
 
     let authToken
     let csrfToken
-    if (typeof authentication === 'object') {
-      if (authentication.mode === AUTHENTICATION_MODES.KEY) {
-        if (!authentication.key) {
-          throw new Error('No key provided for key authentication')
-        }
-        authToken = new Token(authentication.key).get()
-      } else if (authentication.mode === AUTHENTICATION_MODES.SESSION) {
-        if (!authentication.csrfToken) {
-          throw new Error('No CSRF token provided for session authentication')
-        }
-        csrfToken = authentication.csrfToken
+    if (authentication.mode === AUTHENTICATION_MODES.KEY) {
+      if (!authentication.key) {
+        throw new Error('No key provided for key authentication')
+      }
 
+      authToken = new Token(authentication.key).get()
+    } else if (authentication.mode === AUTHENTICATION_MODES.SESSION) {
+      if (!authentication.csrfToken) {
+        throw new Error('No CSRF token provided for session authentication')
       }
-    } else {
-      throw new Error('No authentication settings provided')
-    }
-    this._stackConfig = stackConfig
-    const headers = axiosConfig.headers || {}
 
-    let Authorization = null
-    if (typeof authToken === 'string') {
-      Authorization = `Bearer ${authToken}`
+      csrfToken = authentication.csrfToken
     }
 
     const stackComponents = stackConfig.availableComponents
@ sdk/js/src/api/http.js:61 @ class Http {
         acc[curr] = axios.create({
           baseURL: componentUrl,
           headers: {
-            ...(Authorization ? { Authorization } : {}),
+            ...(typeof authToken === 'string' ? { Authorization: `Bearer ${authToken}` } : {}),
             ...(typeof csrfToken === 'string' ? { 'X-CSRF-Token': csrfToken } : {}),
-            ...headers,
+            ...(axiosConfig.headers || {}),
           },
           ...axiosConfig,
         })
```